configure.ac

```diff
@@ -73,21 +73,17 @@ if test -d ".git" || test -f ".git" ; then :
      #
      GIT_NUM=`git log --pretty=oneline | wc -l | tr -d '[[:space:]]'`
      GIT_RELEASE="${PACKAGE_VERSION}-${GIT_NUM}-${GIT_TAG}"
-
-     A=`git log src/include/ndpi_typedefs.h|wc -l`
-     B=`git log src/include/ndpi_protocol_ids.h|wc -l`
-     C=`git log src/include/ndpi_api.h.in|wc -l`
-     NDPI_API_VERSION=$((A+B+C))
 else
      GIT_RELEASE="${PACKAGE_VERSION}"
      GIT_DATE=`date -u -r CHANGELOG.md`
-     if test -z "$SOURCE_DATE_EPOCH" ; then :
-         SOURCE_DATE_EPOCH=`date +%s`
-     fi
-     NDPI_API_VERSION=`echo $SOURCE_DATE_EPOCH | cut -c7-10`
 fi
 
-NDPI_API_VERSION=`echo $NDPI_API_VERSION | sed 's/^0*//'`
+NDPI_API_VERSION=0x$( cat \
+    src/include/ndpi_typedefs.h \
+    src/include/ndpi_protocol_ids.h \
+    src/include/ndpi_api.h.in \
+    | md5sum | cut -c1-16 \
```

Comment:
Is it better to be extra cautious here and add all the headers in `src/include/`?
For example, `ndpi_define.h.in` and `ndpi_patricia_typedefs.h`are definitely needed, not sure about the others...

Reply:
Am I right that by replacing the time based approach with a checksum based one, the API version is now not constantly increasing? That may break (sooner or later) applications which use nDPI's dev branch for compilation.

Reply:
It always has been ramdom. (by number of lines in files). So I did not make it worse.

Reply:
> Actually, this is not needed because backward-compatibiliy is absent, and also API versions are never compared in the code. (only == and not < or >).


The API version may not be used by libnDPI itself. But other developers may use it for their own applications. The current API version is pretty much obsolete. But making it even more obsolete shouldn't be the goal, instead I would pledge to remove the API version.

Reply:
Yes, removing API version looks OK to me.